lib/state_change_syncer.go

```diff
@@ -455,6 +460,16 @@ func (stateChangeSyncer *StateChangeSyncer) _handleStateSyncerOperation(event *S
 		encoderType = keyEncoder.GetEncoderType()
 		stateChangeEntry.Encoder = keyEncoder
 		stateChangeEntry.EncoderBytes = nil
+
+		if stateChangeEntry.AncestralRecordBytes != nil && len(stateChangeEntry.AncestralRecordBytes) > 0 {
+			// Decode the ancestral record.
+			ancestralRecord, err := DecodeStateKey(stateChangeEntry.KeyBytes, stateChangeEntry.AncestralRecordBytes)
+			if err != nil {
+				glog.Fatalf("Server._handleStateSyncerOperation: Error decoding ancestral record: %v", err)
+			}
+			stateChangeEntry.AncestralRecord = ancestralRecord
+			stateChangeEntry.AncestralRecordBytes = nil
```

Comment:
why do we set `AncestralRecordBytes` to nil?

Reply:
The encoder/decoder here is built to favor the bytes when present instead of the ancestral record. In this instance, we'll be modifying the `AncestralRecord` property down the line, so we want the encoder to encode that `AncestralRecord` field to bytes, rather than just copying the `AncestralRecordBytes`. I've added a comment to explain this in more detail.